Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -259,13 +279,176 @@ public class Klaviyo: NSObject  {
+    /// See https://help.klaviyo.com/hc/en-us/articles/360046055671-Accepted-phone-number-formats-for-SMS-in-Klaviyo
+    /// for information on phone numbers Klaviyo accepts.
+    /// - Parameter phonNumber: a string contining the users phone number.
+    /// - Returns: a KlaviyoSDK instance
+    @_spi(KlaviyoPrivate)
+    @discardableResult
+    public func set(phoneNumber: String) -> KlaviyoSDK {
+        dispatchOnMainThread(action: .setPhoneNumber(phoneNumber))
+        return self
+    }
+    
+    /// Set the current user's external id.
+    /// This could be an id from a system external to Klaviyo, for example your backend's user id.
+    /// NOTE: Please consult with https://help.klaviyo.com/hc/en-us/articles/12902308138011-Understanding-identity-resolution-in-Klaviyo-
+    /// and familiarize yourself with identity resolution before using this identifier.
+    /// - Parameter externalId: a string containing an external id
+    /// - Returns: a KlaviyoSDK instance
+    @_spi(KlaviyoPrivate)
+    @discardableResult
+    public func set(externalId: String) -> KlaviyoSDK {
+        dispatchOnMainThread(action: .setExternalId(externalId))
+        return self
+    }
+    
+    /// Set a profile property on the current user's propfile.
+    /// - Parameter profileAttribute: a profile attribute key to be set on the user's profile.
+    /// - Parameter value: any encodable value profile property value.
+    /// - Returns: a KlaviyoSDK instance
+    @_spi(KlaviyoPrivate)
+    @discardableResult
+    public func set(profileAttribute: Profile.ProfileKey, value: Any) -> KlaviyoSDK  {
+        // This seems tricky to implement with Any - might need to restrict to something equatable, encodable....
+        dispatchOnMainThread(action: .setProfileProperty(profileAttribute, AnyEncodable(value)))
+        return self
+    }
+    
+    /// Create and send an event for the current user.
+    /// - Parameter event: the event to be tracked in Klaviyo
+    @_spi(KlaviyoPrivate)
+    public func create(event: Event) {
+        dispatchOnMainThread(action: .enqueueEvent(event))
+    }
+    
+    /// Set the current user's push token. This will be associated with profile and can be used to send them push notificaitons.
+    /// - Parameter pushToken: data object containing a push token.
+    /// - Returns: a KlaviyoSDK instance
+    @_spi(KlaviyoPrivate)
+    public func set(pushToken: Data) -> KlaviyoSDK {
+        let apnDeviceToken = pushToken.map { String(format: "%02.2hhx", $0) }.joined()
+        dispatchOnMainThread(action: .setPushToken(apnDeviceToken))
+        return self
+    }
+    
+    /// Track a notificationResponse open event in Klaviyo
+    /// - Parameters:
+    ///   - remoteNotification: the remote notificaiton that was opened
+    ///   - fetchCompletionHandler: a completion handler that will be called with a result for Klaviyo notifications
+    /// - Returns: true if the notificaiton originated from Klaviyo, false otherwise.
```

Comment:
should both of these be combined? Like does it makes sense to have both the completion handler and the function return indicate the success of the operation?

Reply:
This completion handler is meant to be the one given to us by the delegate method. If you do not call it you get a warning. However if we aren't handling the callback we should call it. So unfortunately no we can't combine them.

Reply:
ah I thought so.